File: omnic-v2/dist_v3/core/token.py
TOKEN_EOF = 0
TOKEN_ILLEGAL = 1
TOKEN_IDENTIFIER = 10
TOKEN_INT = 11
TOKEN_STRING = 12
TOKEN_ASSIGN = 20
TOKEN_PLUS = 21
TOKEN_MINUS = 22
TOKEN_BANG = 23
TOKEN_ASTERISK = 24
TOKEN_SLASH = 25
TOKEN_LT = 26
TOKEN_GT = 27
TOKEN_EQ = 28
TOKEN_NOT_EQ = 29
TOKEN_COLON = 30
TOKEN_DOT = 31
TOKEN_AND = 32
TOKEN_OR = 33
TOKEN_LE = 34
TOKEN_GE = 35
TOKEN_COMMA = 40
TOKEN_SEMICOLON = 41
TOKEN_LPAREN = 42
TOKEN_RPAREN = 43
TOKEN_LBRACE = 44
TOKEN_RBRACE = 45
TOKEN_LBRACKET = 46
TOKEN_RBRACKET = 47
TOKEN_FN = 60
TOKEN_LET = 61
TOKEN_TRUE = 62
TOKEN_FALSE = 63
TOKEN_IF = 64
TOKEN_ELSE = 65
TOKEN_RETURN = 66
TOKEN_WHILE = 67
TOKEN_STRUCT = 70
TOKEN_NATIVE = 80
TOKEN_IMPORT = 90
TOKEN_PACKAGE = 91
class Token:
    def __init__(self, data=None):
        if data is None: data = {}
        self.kind = data.get('kind')
        self.lexeme = data.get('lexeme')
        self.line = data.get('line')
        self.start = data.get('start')
        self.end = data.get('end')


    def new_token(kind, lexeme, line):
        return Token({ 'kind': kind, 'lexeme': lexeme, 'line': line, 'start': 0, 'end': 0 })

File: omnic-v2/dist_v3/core/token.js
let TOKEN_EOF = 0;
let TOKEN_ILLEGAL = 1;
let TOKEN_IDENTIFIER = 10;
let TOKEN_INT = 11;
let TOKEN_STRING = 12;
let TOKEN_ASSIGN = 20;
let TOKEN_PLUS = 21;
let TOKEN_MINUS = 22;
let TOKEN_BANG = 23;
let TOKEN_ASTERISK = 24;
let TOKEN_SLASH = 25;
let TOKEN_LT = 26;
let TOKEN_GT = 27;
let TOKEN_EQ = 28;
let TOKEN_NOT_EQ = 29;
let TOKEN_COLON = 30;
let TOKEN_DOT = 31;
let TOKEN_AND = 32;
let TOKEN_OR = 33;
let TOKEN_LE = 34;
let TOKEN_GE = 35;
let TOKEN_COMMA = 40;
let TOKEN_SEMICOLON = 41;
let TOKEN_LPAREN = 42;
let TOKEN_RPAREN = 43;
let TOKEN_LBRACE = 44;
let TOKEN_RBRACE = 45;
let TOKEN_LBRACKET = 46;
let TOKEN_RBRACKET = 47;
let TOKEN_FN = 60;
let TOKEN_LET = 61;
let TOKEN_TRUE = 62;
let TOKEN_FALSE = 63;
let TOKEN_IF = 64;
let TOKEN_ELSE = 65;
let TOKEN_RETURN = 66;
let TOKEN_WHILE = 67;
let TOKEN_STRUCT = 70;
let TOKEN_NATIVE = 80;
let TOKEN_IMPORT = 90;
let TOKEN_PACKAGE = 91;
class Token {
    constructor(data = {}) {
        this.kind = data.kind;
        this.lexeme = data.lexeme;
        this.line = data.line;
        this.start = data.start;
        this.end = data.end;
    }
}
function new_token(kind, lexeme, line) {
    return new Token({ kind: kind, lexeme: lexeme, line: line, start: 0, end: 0 });
}

module.exports = { Token, new_token, TOKEN_EOF, TOKEN_ILLEGAL, TOKEN_IDENTIFIER, TOKEN_INT, TOKEN_STRING, TOKEN_ASSIGN, TOKEN_PLUS, TOKEN_MINUS, TOKEN_BANG, TOKEN_ASTERISK, TOKEN_SLASH, TOKEN_LT, TOKEN_GT, TOKEN_EQ, TOKEN_NOT_EQ, TOKEN_COLON, TOKEN_DOT, TOKEN_AND, TOKEN_OR, TOKEN_LE, TOKEN_GE, TOKEN_COMMA, TOKEN_SEMICOLON, TOKEN_LPAREN, TOKEN_RPAREN, TOKEN_LBRACE, TOKEN_RBRACE, TOKEN_LBRACKET, TOKEN_RBRACKET, TOKEN_FN, TOKEN_LET, TOKEN_TRUE, TOKEN_FALSE, TOKEN_IF, TOKEN_ELSE, TOKEN_RETURN, TOKEN_WHILE, TOKEN_STRUCT, TOKEN_NATIVE, TOKEN_IMPORT, TOKEN_PACKAGE };
